linera-service/src/cli/main.rs

```diff
@@ -2139,7 +2139,14 @@ Make sure to use a Linera client compatible with this network.
 
             let clients = (0..*processes)
                 .map(|n| {
-                    let path_provider = PathProvider::new(ssd_dir)?;
+                    let path_provider = if let Some(client_state_dir) = client_state_dir {
+                        let subdir =
+                            PathBuf::from(client_state_dir).join(rand::random::<u16>().to_string());
```

Comment:
`u16` is too short. You'll start getting collisions around 256 entries.

There exist APIs to create unique temporary directories (including in a given folder)